analyzer/consensus/consensus.go

```diff
@@ -275,20 +275,21 @@ func (m *Main) processBlock(ctx context.Context, height int64) error {
 
 	// Prepare and perform updates.
 	batch := &storage.QueryBatch{}
+	queries := make([]*storage.QueryBatch, 6)
 
-	type prepareFunc = func(context.Context, int64, *storage.QueryBatch) error
-	for _, f := range []prepareFunc{
+	type prepareFunc = func(context.Context, int, []*storage.QueryBatch, int64) error
+	for i, f := range []prepareFunc{
 		m.prepareBlockData,
 		m.prepareRegistryData,
 		m.prepareStakingData,
 		m.prepareSchedulerData,
 		m.prepareGovernanceData,
 	} {
-		func(f prepareFunc) {
+		func(f prepareFunc, i int) {
 			group.Go(func() error {
-				return f(groupCtx, height, batch)
+				return f(groupCtx, i, queries, height)
```

Comment:
looks like these routines don't interact with other indices, maybe we could do something like

```go
queries := make([]storage.QueryBatch, 6)
...
f(groupCtx, height, &queries[i])
```

Reply:
so that we don't have to change the function signatures

Reply:
Updated; thanks!